src/lib/ikea.ts

```diff
@@ -825,3 +838,87 @@ export function addCustomClusterManuSpecificIkeaUnknown(): ModernExtend {
         commandsResponse: {},
     });
 }
+
+const unfreezeMechanisms: {
+    [key: string]: (entity: Zh.Endpoint | Zh.Group) => Promise<void>;
+} = {
+    // WS lights:
+    //   Aborts the color transition midway: light will stay at the intermediary
+    //   state it was when it received the command.
+    // Color lights:
+    //   Do not support this command.
+    moveColorTemp: async (entity) => {
+        await entity.command("lightingColorCtrl", "moveColorTemp", {rate: 1, movemode: 0, minimum: 0, maximum: 600}, {});
+    },
+
+    // WS lights:
+    //   Same as "moveColorTemp".
+    // Color lights:
+    //   Finishes the color transition instantly: light will instantly
+    //   "fast forward" to the final state, post-transition.
+    genLevelCtrl: async (entity) => {
+        await entity.command("genLevelCtrl", "stop", {}, {});
+    },
+};
+
+// zigbee commands which will freeze an IKEA light
+const willFreeze = (clusterKey: string | number, payload: KeyValue): payload is KeyValue & {transtime: number} =>
+    payload &&
+    // any color command with a transition will freeze the light...
+    (payload.transtime as number) > 0 &&
+    clusterKey === "lightingColorCtrl" &&
+    // ...except for 'stop' commands:
+    payload.rate !== 1 &&
+    payload.movemode !== 0;
+
+// keys for whom we need to unfreeze the light before issuing a corresponding command to said light
+const keysNeedingUnfreeze = new Set(["state", "brightness", "brightness_percent", "color_temp", "color_temp_percent"]);
+
+const ikea_bulb_unfreeze = (next: Tz.Converter["convertSet"]) => {
+    const converter: Tz.Converter["convertSet"] = async (entity, key, value, meta) => {
+        if (!keysNeedingUnfreeze.has(key)) {
+            return await next(entity, key, value, meta);
+        }
+
+        const id = "deviceIeeeAddress" in entity ? entity.deviceIeeeAddress : entity.groupID;
+
+        const proxyCommand: Zh.Endpoint["command"] = async (clusterKey, commandKey, payload, options) => {
```

Comment:
Can't we extend/override the toZigbee converter instead of changing the `command` here? I think that will be a more maintainable solution

Reply:
I agree it'd be more maintainable - the issue is that we want to detect when we're sending a `lightingColorCtrl` as the cluster key, which I don't think we can do at the toZigbee level (I may be wrong)

Happy to go with your request still, in which case we would pessimise the code and assume keys such as `color_temp`, `brightness` will generate `lightingColorCtrl`, and use those (instead of `lightingColorCtrl`) to detect if a bulb will be frozen

Reply:
`color_temp` will always generate a `lightingColorCtrl` right? `brightness` will always be `genLevelCtrl`, note that `state` can also sometimes generate a `genLevelCtrl` (in case a default transition is specified for the bulb)

Reply:
Sure, I've updated to go off those keys instead: cdd887ff5f2 (the large hunk is mostly whitespace)